desc/efit.py

```diff
@@ -0,0 +1,374 @@
+
+                def splitline(s):
+                    return s.split()
+
+                def splitarr(l):
+                    return " ".join(l).split()
+
+            else:
+
+                def splitline(s):
+                    ss = s.split("\n")[0]
+                    idx = np.arange(0, len(ss) + 1, 16)
+                    return [ss[i:j] for i, j in zip(idx[:-1], idx[1:])]
+
+                def splitarr(l):
+                    return splitline("".join([line.split("\n")[0] for line in l]))
+
+            line0 = lines[0].split()
+
+            # points in the horizontal direction (width) and vertical direction (height)
+            g["nw"] = int("".join([c for c in line0[-2] if c.isdigit()]))
+            g["nh"] = int("".join([c for c in line0[-1] if c.isdigit()]))
+            if len(line0) == 8:
+                g["efit"] = line0[4]
+            else:
+                g["efit"] = None
+
+            [g["rdim"], g["zdim"], g["rcentr"], g["rleft"], g["zmid"]] = [
+                float(foo) for foo in splitline(lines[1])
+            ]
+            [
+                g["rmaxis"],
+                g["zmaxis"],
+                g["chi_axis"],
+                g["chi_boundary"],
+                g["bcentr"],
+            ] = [float(foo) for foo in splitline(lines[2])]
+            [g["zmaxis"], _, g["chi_boundary"], _, _] = [
+                float(foo) for foo in splitline(lines[4])
+            ]
+            lines_per_profile = int(np.ceil(g["nw"] / 5))
+            lines_psirz = int(np.ceil(g["nw"] * g["nh"] / 5))
+            lines = lines[5:]
+
+            # All the flux functions are stored in first few rows of the file
+            g["fpol"] = np.array(
+                [float(foo) for foo in splitarr(lines[:lines_per_profile])]
+            )
+            lines = lines[
+                lines_per_profile:
+            ]  # redefine lines to remove the lines we just read
+
+            g["pres"] = np.array(
+                [float(foo) for foo in splitarr(lines[:lines_per_profile])]
+            )
+            lines = lines[lines_per_profile:]
+
+            g["ffprime"] = np.array(
+                [float(foo) for foo in splitarr(lines[:lines_per_profile])]
+            )
+            lines = lines[lines_per_profile:]
+
+            g["pprime"] = np.array(
+                [float(foo) for foo in splitarr(lines[:lines_per_profile])]
+            )
+            lines = lines[lines_per_profile:]
+
+            # Read the poloidal flux stored as a 2D array
+            g["chirz"] = np.array(
+                [float(foo) for foo in splitarr(lines[:lines_psirz])]
+            ).reshape((g["nh"], g["nw"]))
+            lines = lines[lines_psirz:]
+
+            # Safety factor
+            g["q"] = np.array(
+                [float(foo) for foo in splitarr(lines[:lines_per_profile])]
+            )
+            lines = lines[lines_per_profile:]
+
+            if bdry_dist == 1:  # distance of the boundary from the axis
```

Comment:
what is this distance in? some normalized unit? $\rho$ or $\psi_n$?

Reply:
It's normalized poloidal flux. Adding that to the docstring.